src/ImageSharp/Formats/Tiff/Compression/Compressors/DeflateCompressor.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.IO;
+using SixLabors.ImageSharp.Compression.Zlib;
+using SixLabors.ImageSharp.Formats.Experimental.Tiff.Constants;
+using SixLabors.ImageSharp.Memory;
+
+namespace SixLabors.ImageSharp.Formats.Experimental.Tiff.Compression.Compressors
+{
+    internal class DeflateCompressor : TiffBaseCompressor
+    {
+        private readonly DeflateCompressionLevel compressionLevel;
+
+        private readonly MemoryStream memoryStream = new MemoryStream();
+
+        public DeflateCompressor(Stream output, MemoryAllocator allocator, int width, int bitsPerPixel, TiffPredictor predictor, DeflateCompressionLevel compressionLevel)
+            : base(output, allocator, width, bitsPerPixel, predictor)
+            => this.compressionLevel = compressionLevel;
+
+        public override TiffEncoderCompression Method => TiffEncoderCompression.Deflate;
+
+        public override void Initialize(int rowsPerStrip)
+        {
+        }
+
+        public override void CompressStrip(Span<byte> rows, int height)
+        {
+            this.memoryStream.Seek(0, SeekOrigin.Begin);
+            using var stream = new ZlibDeflateStream(this.Allocator, this.memoryStream, this.compressionLevel);
+
+            if (this.Predictor == TiffPredictor.Horizontal)
+            {
+                HorizontalPredictor.ApplyHorizontalPrediction(rows, this.BytesPerRow, this.BitsPerPixel);
+            }
+
+            stream.Write(rows);
+
+            stream.Flush();
+            //// stream.Dispose(); // todo: dispose write crc
```

Comment:
the stream should really be disposed before getting the size, otherwise the crc will not be written as the comment states.
I wonder how this actually works without it.

Reply:
I think that readers are ignored the `CRC`, I'll add crc writing.